common/include/pcl/common/impl/file_io.hpp

```diff
@@ -42,29 +42,21 @@
 namespace pcl
 {
 
-#ifndef _WIN32
   void getAllPcdFilesInDirectory(const std::string& directory, std::vector<std::string>& file_names)
   {
-    DIR *dp;
-    struct dirent *dirp;
-    if((dp  = opendir(directory.c_str())) == nullptr) {
-      std::cerr << "Could not open directory.\n";
-      return;
+    boost::filesystem::path p(directory);
+    if(boost::filesystem::is_directory(p)) {
+      for(auto& entry : boost::make_iterator_range(boost::filesystem::directory_iterator(p), {}))
+        if (boost::filesystem::is_regular_file(entry)) {
```

Comment:
You could use `Boost.filesystem.path.extension() == "pcd"` and the valid symlink test here itself.

Reply:
So do you want me to resolved symlinks to check if they are a directory or if it is a regular file?

Reply:
No. Symlinks can be broken (file pointed to doesn't exist). Those aren't readable down-the-line

Reply:
> You could use `Boost.filesystem.path.extension() == "pcd"` and the valid symlink test here itself.

Incorporated  in 41f839d
Any other changes needed in this?

Reply:
No